kodi_addon_checker/report.py

```diff
@@ -0,0 +1,80 @@
+import inspect
+from abc import ABC
+
+PROBLEM = "ERROR"
+WARNING = "WARN"
+INFORMATION = "INFO"
+
+KODI_REPORTERS = {}
+
+
+class Report(object):
+    def __init__(self, artifact_name):
+        self.artifact_name = artifact_name
+        self.problem = 0
+        self.warning = 0
+        self.information = 0
+        self.reports = []
+
+    def log(self, report):
+        self.reports.append(report)
+        self.problem += report.problem
+        self.warning += report.warning
+        self.information += report.information
+
+    def __iter__(self):
+        return iter(self.reports)
+
+
+class Record(Report):
+    def __init__(self, artifact_name, message, start_line=-1, end_line=-1, start_char_position=-1,
+                 end_char_position=-1):
+        super().__init__(artifact_name)
+        self.message = message
+        self.start_line = start_line
+        self.end_line = end_line
+        self.start_char_position = start_char_position
+        self.end_char_position = end_char_position
+
+        if PROBLEM == artifact_name:
```

Comment:
I don't think we should have fields for problem counts (etc) directly on the record type. Move this logic to the report log method. And get rid of these fields here.

Reply:
Fixed